spec/unit/typesafe/config/conf_parser_spec.rb

```diff
@@ -59,3 +63,760 @@ def parse(s)
+
+    # should have loaded conf, json... skipping properties
+    expect(conf.get_int("ints.fortyTwo")).to eq(42)
+    expect(conf.get_int("fromJson1")).to eq(1)
+  end
+
+  it "includeFileWithExtension" do
+    conf = Hocon::ConfigFactory.parse_string("include file(" +
+              TestUtils.json_quoted_resource_file("test01.conf") + ")")
+
+    expect(conf.get_int("ints.fortyTwo")).to eq(42)
+    expect(conf.has_path?("fromJson1")).to eq(false)
+    expect(conf.has_path?("fromProps.abc")).to eq(false)
+  end
+
+  it "includeFileWhitespaceInsideParens" do
+    conf = Hocon::ConfigFactory.parse_string("include file(  \n  " +
+              TestUtils.json_quoted_resource_file("test01") + "  \n  )")
+
+    # should have loaded conf, json... NOT properties
+    expect(conf.get_int("ints.fortyTwo")).to eq(42)
+    expect(conf.get_int("fromJson1")).to eq(1)
+  end
+
+  it "includeFileNoWhitespaceOutsideParens" do
+    e = TestUtils.intercept(Hocon::ConfigError::ConfigParseError) {
+      Hocon::ConfigFactory.parse_string("include file (" +
+        TestUtils.json_quoted_resource_file("test01") + ")")
+    }
+    expect(e.message.include?("expecting include parameter")).to eq(true)
+  end
+
+  it "includeFileNotQuoted" do
+    # this test cannot work on Windows
+    f = TestUtils.resource_file("test01")
+    if (f.to_s.include?("\\"))
+      $stderr.puts("includeFileNotQuoted test skipped on Windows")
+    else
+      e = TestUtils.intercept(Hocon::ConfigError::ConfigParseError) {
+        Hocon::ConfigFactory.parse_string("include file(" + f + ")")
+      }
+      expect(e.message.include?("expecting include parameter")).to eq(true)
+    end
+  end
+
+  it "includeFileNotQuotedAndSpecialChar" do
+    f = TestUtils.resource_file("test01")
+    if (f.to_s.include?("\\"))
+      $stderr.puts("includeFileNotQuoted test skipped on Windows")
+    else
+      e = TestUtils.intercept(Hocon::ConfigError::ConfigParseError) {
+        Hocon::ConfigFactory.parse_string("include file(:" + f + ")")
+      }
+      expect(e.message.include?("expecting a quoted string")).to eq(true)
+    end
+
+  end
+
+  it "includeFileUnclosedParens" do
+    e = TestUtils.intercept(Hocon::ConfigError::ConfigParseError) {
+      Hocon::ConfigFactory.parse_string("include file(" + TestUtils.json_quoted_resource_file("test01") + " something")
+    }
+    expect(e.message.include?("expecting a close paren")).to eq(true)
+  end
+
+  # Skipping 'includeURLBasename' because we don't support URLs
+  # Skipping 'includeURLWithExtension' because we don't support URLs
+  # Skipping 'includeURLInvalid' because we don't support URLs
+  # Skipping 'includeResources' because we don't support classpath resources
+  # Skipping 'includeURLHeuristically' because we don't support URLs
+  # Skipping 'includeURLBasenameHeuristically' because we don't support URLs
+
+  it "acceptBOMStartingFile" do
+    skip("BOM not parsing properly yet") do
```

Comment:
Sure, using the `skip` string is probably better than a comment.  Maybe just add what you said to me in person - something like "BOM is not parsing properly and we don't care about this because it only affects Windows" or something along those lines.